libcnb/src/exec_d.rs

```diff
@@ -0,0 +1,25 @@
+use libcnb_data::exec_d::ExecDProgramOutput;
+use std::fs::File;
+use std::io::BufWriter;
+use std::io::Write;
+
+/// Writes the output of a CNB exec.d program in a spec compliant way.
+pub fn write_exec_d_program_output<O: Into<ExecDProgramOutput>>(o: O) {
+    // Allow compilation of exec.d programs under windows, but fail at runtime:
+    #[cfg(target_family = "windows")]
+    unimplemented!("libcnb.rs does not support running in Windows containers yet!");
+
+    #[cfg(target_family = "unix")]
```

Comment:
I'm not sure what the most idiomatic form here is, but this does not seem to be overly complex and it works. I move the attributes in the body of the function to avoid documentation duplication and only conditionally run the platform specific parts.

The intent is very similar to your proposal, the only difference is that you split it into functions and I went with statements in the body of the function.

This feels a bit nitpicky and I don't feel strongly about this. If you have a strong opinion that functions would be better I'm fine with it - but I don't see an issue with how it's written currently - especially since there is no real windows support anyway.

Reply:
I just came to me that windows support would require refactoring of this anyway since not only the way we obtain the `File` is different. Windows exec.d programs have arguments whereas the unix ones don't. If we'd ever add support for Windows containers, this file would look very different for sure. :)